### DISCUSSION TITLE
Programmatically  generating yml from pulumi output.

### DISCUSSION THREAD

**Discussion by hsblhsn** [pulumi/pulumi#8414] [Pulumi in Practice] [answered]
Hello there,

I am trying out pulumi. Previously I was using terraform. What I am trying to do here is creating a redis instance and getting it's `auth_string` as output and encode it to yml with other data. That yml will be stored to gcp secret manager.

A demo yml data:

```yml
appName: demo
redisUrl: redis://:secret_auth_str_from_pulumi_output@host_from_pulumi:6379
```

I am using Go and GCP. Is there any way I can solve my problem by code? I can get the output and generate yml using pulumi cli `pulumi stack output ...`. But I want to do it in code. I can't marshal `pulumi.StringOutput` to yml. I need help.

**Comment by jaxxstorm** [comment, marked as answer]
Hey there. you can do this using an apply.

There's an example of how you'd do this here:
https://github.com/pulumi/examples/blob/ca40203279f393c0c159dadcadc97c6007122997/aws-go-assume-role/create-role/main.go#L33-L49

here's a blog post explaining how outputs and apply works:

https://www.leebriggs.co.uk/blog/2021/05/09/pulumi-apply.html
